client/fcidrClient.sp

```diff
@@ -0,0 +1,110 @@
+#include "SCR-Version"
+
+#include <sourcemod>
+#include <socket>
+#include <bytebuff>
+
+#define PROTOCOL_VERSION 1
+
+
+/**
+ * Base message structure
+ * 
+ */
+methodmap Header < ByteBuffer
+{
+	public Header()
+	{
+		return view_as<Header>(CreateByteBuffer());
+	}
+
+	public void baseHeader()
+	{
+		this.WriteByte(PROTOCOL_VERSION);
+	}
+
+	public void Dispatch()
+	{
+		char sDump[MAX_BUFFER_LENGTH];
+
+		int iLen = this.Dump(sDump, MAX_BUFFER_LENGTH);
+
+		this.Close();
+
+		if (!SocketIsConnected(g_hSocket))
+			return;
+
+		// Len required
+		// If len is not included, it will stop at the first \0 terminator
+		SocketSend(g_hSocket, sDump, iLen);
+	}
+}
+
+methodmap Request < Header
+{
+	public Request(int ip_address, StringMap() keyvals)

```

Comment:
keyvals type here would just be `StringMap` instead of `StringMap()` which is a constructor caller.